composer/trainer/trainer.py

```diff
@@ -950,7 +950,9 @@ def eval(self, is_batch: bool):
             self.engine.run_event(Event.EVAL_START)
 
             for evaluator in state.evaluators:
-                dataloader = evaluator.dataloader.dataloader
+                dataloader = evaluator.dataloader
+                if not isinstance(dataloader, ComposerDataLoader):
+                    dataloader = ComposerDataLoader(dataloader)
```

Comment:
I'm not a huge fan of this cast, but we can't do it in the evaluator because that's in core and `ComposerDataLoader` is not.

However, I'm hoping that in a nearby release we can drop the uses of `ComposerDataLoader`-specific fields from the below.

Reply:
The trouble with that is that `Evaluator` is in core, but it can't import `ComposerDataLoader` since that's been moved to `datasets`. I defend `ComposerDataLoader` being moved out of core, since its logic is only relevant to the Composer Trainer.